src/main/java/org/spongepowered/common/mixin/core/server/MixinMinecraftServer.java

```diff
@@ -531,6 +530,33 @@ protected void prepareSpawnArea(WorldServer world) {
     }
 
     @Override
+    public Optional<World> copyWorld(World originalWorld, String copyName) {
+        try {
+            FileUtils.copyDirectory(getWorldDir(originalWorld.getName()), getWorldDir(copyName));
```

Comment:
Actually I think we can do better. Looking at how blocks are marked as dirty for clients to receive the new block changes, world.worldAcesses is a list of observers of the world. If you are doing asynchronous copying creating and registering an IWorldAccess would allow you to receive any updates that happened since the copy was started.

That said I'm not sure which way would be better give the plugin requesting a copy an exact copy at the time it ended, a partial possibly incomplete and out of sync copy.

Neither of those really seem ideal when they more then likely want a copy at the time that the API was called. But having a complete sync'd copy at the time that the future returns seems better then having an out of sync copy that could cause item/entity duplication.